File: dynamic_routing_menu_backend/src/main/java/org/example/entity/Menu.java
package org.example.entity;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Getter;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@Getter
@TableName(value = "system_menu_table")
public class Menu implements Serializable {

   @TableId(type = IdType.INPUT)
   private Long menuId;
   private String menuName;
   private String menuPath;
   private String menuIcon;
   private String menuComponent;
   private String menuTitle;
   private Long parentId;
   private Boolean isDirectory;

   @TableField(exist = false)
   private List<Menu> children = new ArrayList<>();

}

File: dynamic_routing_menu_front/src/components/menu/menu.ts

// 定义菜单列表
export interface IMenuList {
    
  //id: number;
  menuId: number;
  menuName: string;
  menuIcon: string;
  menuPath: string;
  menuTitle: string;
  parentId: number;
  menuComponent : string;
  isDirectory?: boolean;
}

// 定义菜单（树）项接口
export interface IMenuItem extends IMenuList {
  
  children: IMenuItem[];

}

export interface ICurrentMenuItem extends IMenuList{
  //isExisted: boolean;
}

export const tempMenuList: IMenuList = {
      //id: 0,
      menuId: 0,
      menuName: "",
      menuIcon: "",
      menuPath: "",
      menuTitle: "",
      parentId: -1,
      menuComponent: ""
};


export const initTempMenuList = (list: IMenuList):IMenuList => {
        
        list.menuId = 0;
        list.menuName = "";
        list.menuIcon = "";
        list.menuPath = "";
        list.menuTitle = "";
        list.parentId = -1;
        list.menuComponent = "";
        
        return list;
}